### PR TITLE
Adds direct workflow Allegro hand environment

### PR DESCRIPTION
**Pull request by kellyguo11**
# Description
This change refactors the shadow hand environment into a general in-hand manipulation environment and extends it with the Allegro hand.

## Type of change

- New feature (non-breaking change which adds functionality)

## Checklist

- [x] I have run the [`pre-commit` checks](https://pre-commit.com/) with `./isaaclab.sh --format`
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have updated the changelog and the corresponding version in the extension's `config/extension.toml` file
- [x] I have added my name to the `CONTRIBUTORS.md` or my name already exists there

### CODE REVIEW

File: source/extensions/omni.isaac.lab_tasks/omni/isaac/lab_tasks/direct/allegro_hand/allegro_hand_env_cfg.py [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+        physics_material=RigidBodyMaterialCfg(
+            static_friction=1.0,
+            dynamic_friction=1.0,
+        ),
+        physx=PhysxCfg(
+            bounce_threshold_velocity=0.2,
+        ),
+    )
+    # robot
+    robot_cfg: ArticulationCfg = ALLEGRO_HAND_CFG.replace(prim_path="/World/envs/env_.*/Robot").replace()
```
Comment (Dhoeller19) [inline source/extensions/omni.isaac.lab_tasks/omni/isaac/lab_tasks/direct/allegro_hand/allegro_hand_env_cfg.py]:
```suggestion
    robot_cfg: ArticulationCfg = ALLEGRO_HAND_CFG.replace(prim_path="/World/envs/env_.*/Robot")
```

File: docs/source/features/environments.rst [hunk 2]
```diff
@@ -98,10 +99,11 @@ for the reach environment:
 .. |lift-cube-ik-rel-link| replace:: `Isaac-Lift-Cube-Franka-IK-Rel-v0 <https://github.com/isaac-sim/IsaacLab/blob/main/source/extensions/omni.isaac.lab_tasks/omni/isaac/lab_tasks/manager_based/manipulation/lift/config/franka/ik_rel_env_cfg.py>`__
 .. |cabi-franka-link| replace:: `Isaac-Open-Drawer-Franka-v0 <https://github.com/isaac-sim/IsaacLab/blob/main/source/extensions/omni.isaac.lab_tasks/omni/isaac/lab_tasks/manager_based/manipulation/cabinet/config/franka/joint_pos_env_cfg.py>`__
 .. |cube-allegro-link| replace:: `Isaac-Repose-Cube-Allegro-v0 <https://github.com/isaac-sim/IsaacLab/blob/main/source/extensions/omni.isaac.lab_tasks/omni/isaac/lab_tasks/manager_based/manipulation/inhand/config/allegro_hand/allegro_env_cfg.py>`__
+.. |allegro-direct-link| replace:: `Isaac-Allegro-Hand-Direct-v0 <https://github.com/isaac-sim/IsaacLab/blob/main/source/extensions/omni.isaac.lab_tasks/omni/isaac/lab_tasks/direct/allegro_hand/allegro_hand_env_cfg.py>`__
```
Comment (Dhoeller19) [inline docs/source/features/environments.rst]:
```suggestion
.. |allegro-direct-link| replace:: `Isaac-Repose-Cube-Allegro-Direct-v0 <https://github.com/isaac-sim/IsaacLab/blob/main/source/extensions/omni.isaac.lab_tasks/omni/isaac/lab_tasks/direct/allegro_hand/allegro_hand_env_cfg.py>`__
```

File: source/extensions/omni.isaac.lab_tasks/omni/isaac/lab_tasks/direct/allegro_hand/__init__.py [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+from . import agents
+from .allegro_hand_env_cfg import AllegroHandEnvCfg
+
+##
+# Register Gym environments.
+##
+
+gym.register(
+    id="Isaac-Allegro-Hand-Direct-v0",
```
Comment (Dhoeller19) [inline source/extensions/omni.isaac.lab_tasks/omni/isaac/lab_tasks/direct/allegro_hand/__init__.py]:
```suggestion
    id="Isaac-Repose-Cube-Allegro-Direct-v0",
```

File: source/extensions/omni.isaac.lab_tasks/omni/isaac/lab_tasks/direct/inhand_manipulation/inhand_manipulation_env.py [hunk 4]
```diff
@@ -12,18 +12,16 @@
...
-from .shadow_hand_env_cfg import ShadowHandEnvCfg
 
+class InHandManipulationEnv(DirectRLEnv):
+    cfg: DirectRLEnvCfg
 
-class ShadowHandEnv(DirectRLEnv):
-    cfg: ShadowHandEnvCfg
-
-    def __init__(self, cfg: ShadowHandEnvCfg, render_mode: str | None = None, **kwargs):
+    def __init__(self, cfg: DirectRLEnvCfg, render_mode: str | None = None, **kwargs):
```
Comment (Dhoeller19) [inline source/extensions/omni.isaac.lab_tasks/omni/isaac/lab_tasks/direct/inhand_manipulation/inhand_manipulation_env.py]:
```suggestion
from omni.isaac.lab_tasks.direct.allegro_hand import AllegroHandEnvCfg
from omni.isaac.lab_tasks.direct.shadow_hand import ShadowHandEnvCfg


class InHandManipulationEnv(DirectRLEnv):
    cfg: AllegroHandEnvCfg | ShadowHandEnvCfg

    def __init__(self, cfg: AllegroHandEnvCfg | ShadowHandEnvCfg, render_mode: str | None = None, **kwargs):
```

File: source/extensions/omni.isaac.lab_tasks/omni/isaac/lab_tasks/direct/inhand_manipulation/inhand_manipulation_env.py [hunk 5]
```diff
@@ -12,18 +12,16 @@
 
 import omni.isaac.lab.sim as sim_utils
 from omni.isaac.lab.assets import Articulation, RigidObject
-from omni.isaac.lab.envs import DirectRLEnv
+from omni.isaac.lab.envs import DirectRLEnv, DirectRLEnvCfg
```
Comment (Dhoeller19) [inline source/extensions/omni.isaac.lab_tasks/omni/isaac/lab_tasks/direct/inhand_manipulation/inhand_manipulation_env.py]:
```suggestion
from omni.isaac.lab.envs import DirectRLEnv
```

File: source/extensions/omni.isaac.lab_tasks/omni/isaac/lab_tasks/direct/inhand_manipulation/inhand_manipulation_env.py [hunk 6]
```diff
@@ -281,10 +279,10 @@ def compute_reduced_observations(self):
         #   Relative target orientation
         obs = torch.cat(
             (
-                self.fingertip_pos.view(self.num_envs, self.num_fingertips * 3),  # 0:15
-                self.object_pos,  # 15:18
-                quat_mul(self.object_rot, quat_conjugate(self.goal_rot)),  # 18:22
-                self.actions,  # 22:42
+                self.fingertip_pos.view(self.num_envs, self.num_fingertips * 3),  # 0:15 (SH)
```
Comment (Dhoeller19) [inline source/extensions/omni.isaac.lab_tasks/omni/isaac/lab_tasks/direct/inhand_manipulation/inhand_manipulation_env.py]: Shall we remove all the (SH) comments?

File: docs/source/features/environments.rst (hunk 2 above)
Comment (kellyguo11) [reply docs/source/features/environments.rst]: should we rename the shadow hand environment as well then? seems a bit less intuitive to have Isaac-Shadow-Hand and Isaac-Repose-Cube-Allegro when they are pretty much the same task.

Comment (Dhoeller19) [reply docs/source/features/environments.rst]: Yes we should rename the other as well, I was just thinking about backward compatibility, but I guess it is more important to be consistent here.

**Review by Dhoeller19 — APPROVED** [review]

**Pull request merged**